src/wasm-lib/kcl/src/parsing/ast/execute.rs

```diff
@@ -358,8 +358,44 @@ async fn inner_execute_pipe_body(
 }
 
 impl Node<CallExpressionKw> {
-    pub async fn execute(&self, _exec_state: &mut ExecState, _ctx: &ExecutorContext) -> Result<KclValue, KclError> {
-        todo!()
+    #[async_recursion]
+    pub async fn execute(&self, exec_state: &mut ExecState, ctx: &ExecutorContext) -> Result<KclValue, KclError> {
+        let fn_name = &self.callee.name;
+
+        // Build a hashmap from argument labels to the final evaluated values.
+        let mut fn_args = HashMap::with_capacity(self.arguments.len());
+        for arg_expr in &self.arguments {
+            let source_range = SourceRange::from(arg_expr.arg.clone());
+            let metadata = Metadata { source_range };
+            let value = ctx
+                .execute_expr(&arg_expr.arg, exec_state, &metadata, StatementKind::Expression)
+                .await?;
+            fn_args.insert(arg_expr.label.name.clone(), Arg::new(value, source_range));
+        }
+        let fn_args = fn_args; // remove mutability
+
+        // Evaluate the unlabeled first param, if any exists.
+        let unlabeled = if let Some(ref arg_expr) = self.unlabeled {
+            let source_range = SourceRange::from(arg_expr.clone());
+            let metadata = Metadata { source_range };
+            let value = ctx
+                .execute_expr(arg_expr, exec_state, &metadata, StatementKind::Expression)
+                .await?;
+            Some(Arg::new(value, source_range))
+        } else {
+            None
+        };
+
+        let args = crate::std::Args::new_kw(fn_args, unlabeled, self.into(), ctx.clone());
+        match ctx.stdlib.get_either(fn_name) {
+            FunctionKind::Core(func) => {
+                // Attempt to call the function.
+                let mut result = func.std_lib_fn()(exec_state, args).await?;
+                tag_2d_to_3d(&mut result, exec_state)?;
+                Ok(result)
+            }
+            _ => todo!(),
```

Comment:
Can we add a meaningful error message? Also, I would not use `_` so that we can find all references the variants and allow the compiler to help us when adding new variants.